### DISCUSSION TITLE
Allow `node:` imports on Deno

### DISCUSSION THREAD

**Discussion by Tc-001** [withastro/roadmap#550] [Proposal]
### Body

# Summary

Starting from deno 1.28 it is possible to import node's built-in modules (or at least a polyfill of them) using the `node:` specifier the same way that Astro does.

https://deno.land/manual@v1.32.3/node/node_specifiers

This means that theoretically SSR on Deno should be able to support `import fs from "node:fs"` if those imports were left untouched in the build process.

It is also possible to import NPM packages using `npm:<name>` that also does some magic to make them work, but that may be out of scope and already handled by vite.

Only issue could be that currently `node:` and `npm:` imports are not supported in `deno bundle` and `deno compile` etc. 
But that will be fixed once `deno pack` gets implemented, and is not needed to run the code.

# Background & Motivation

This would be great for feature parity, and would make moving from node's SSR to the more modern and secure [citation needed] Deno runtime, as there would be almost zero modifications needed to existing code.

# Goals

A **concise, bulleted-list** outlining the intended goals of this RFC. 

- Support `node:` imports on Deno
- Define a minimum version of Deno and throw a warning if earlier 

# Example

The snippet below works on both Astro and Deno > 1.28
```js
import crypto from "node:crypto"

const hash = crypto
    .createHash('sha256')
    .update("aaaaa")
    .digest('hex');
```

**Comment by florian-lefebvre** [comment]
The deno adapter is now maintained by the Deno team: https://github.com/denoland/deno-astro-adapter
